### PR TITLE
fix_dsi_studio_trk

### PR DESCRIPTION
**Pull request by frheault**
After some time I gathered enough DSI-Studio corner case data to find exactly how to fix them.

Its working perfectly on 1 datasets from Leon C., 1 from Kurt S., HCP, old surgery case from Sherbrooke and the TractEM (BLSA) data
There is at least 2 thing wrong with the TRK header, 2 thing wrong with the TRK data, 1 thing wrong with the NIFTI and the NIFTI is moved to AC-PC.
That 4 things to fix to get it aligned with itself in MI-Brain and 2 more step to get to "native space'' (including rigid registration).

I can't garantee its perfect, but I think its close to it. If the data does not have too much skull and voxels smaller than 2mm iso, its pretty robust.

In summary:

1. Read the tractogram with nibabel API
2. Shift the origin by 1 voxel in X and Y
3. Flip tractogram in X/Y around its center
4. Multiply this by the affine of the DSI nifti, now this is a valid RASMM trk that you can save and it will be lined up with the DSI nifti.

If you want it on top of the original

1. If the native was already LPS, fine. If not, re-flip the TRK & the NIFTI itself around its center for each axis not being L/P/S
2. If the data has a lot of skull, crop a medium cube of from the center and register it (rigid) to the native image
3. Apply transfo to tractogram, cut invalid streamlines and save

### CODE REVIEW

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

File: scripts/scil_fix_dsi_studio_trk.py [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                              'bounding box.')
+    invalid.add_argument('--keep_invalid', action='store_true',
+                         help='Keep the streamlines landing out of the '
+                              'bounding box.')
+    add_overwrite_arg(p)
+
+    return p
+
+
+def get_axis_shift_vector(flip_axes):
```
Comment (arnaudbore) [inline scripts/scil_fix_dsi_studio_trk.py:90]: why do you have this here instead of transformation.py

File: scripts/tests/test_fix_dsi_studio_trk.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import os
+import tempfile
+
+from scilpy.io.fetcher import fetch_data, get_home, get_testing_files_dict
+
+# If they already exist, this only takes 5 seconds (check md5sum)
+fetch_data(get_testing_files_dict(), keys=['tracking.zip'])
```
Comment (arnaudbore) [inline scripts/tests/test_fix_dsi_studio_trk.py:10]: don't need to fetch anything or maybe you could add some dsi data in this package

File: scripts/scil_fix_dsi_studio_trk.py (hunk 1 above)
Comment (arnaudbore) [reply scripts/scil_fix_dsi_studio_trk.py:90]: get_axis_flip_vector

**Review by arnaudbore — APPROVED** [review]
LGTM but I'm not really the one who could test this.

File: scripts/tests/test_fix_dsi_studio_trk.py (hunk 2 above)
Comment (frheault) [reply scripts/tests/test_fix_dsi_studio_trk.py:10]: I deleted the line about the fetch, this script is (for now) for case-by-case basis and DSI data is basically unique so I think for now I would prefer just to keep it like that until someone actually test it in real life condition.

File: scripts/scil_fix_dsi_studio_trk.py (hunk 1 above)
Comment (frheault) [reply scripts/scil_fix_dsi_studio_trk.py:90]: It is a piece of code unique to DSI data correction and there is no use for it outside (for now). Since this is risky spatial transform I would leave it there.

Also it use the flip_axes variable, but it is used as a shift vector (in a -= statement below)

**Comment by scil-admin** [comment]
Build passed ! Good Job :beers: !

**Review by arnaudbore — APPROVED** [review]
